File: backend/app/core/upgrades/stat_modifiers/player_health_3.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class PlayerHealth3(UpgradeImplementation):
    id = 'player_health_3'
    name = 'Reinforced Core'
    description = '+60 max health'
    rarity = 'rare'
    type = 'stat_modifier'
    cost = 10

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 60
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/stat_modifiers/player_health_3.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const PlayerHealth3Def: UpgradeDef = {
  id: "player_health_3",
  name: "Reinforced Core",
  description: "+60 max health",
  rarity: RarityID.Rare,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 10,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: 60,
  stackable: true,
  maxStacks: 99999999,
}

export class PlayerHealth3 extends Upgrade {}
